util/src/network/handshake.rs

```diff
@@ -151,7 +150,9 @@ impl Handshake {
 						try!(self.read_ack_eip8(host.secret(), &data));
 					};
 				},
-				HandshakeState::StartSession => {},
+				HandshakeState::StartSession => {
+					io.clear_timer(self.connection.token).ok();
```

Comment:
Handshake timeout needs to be removed when the handshake is over, not when something is received.